### ISSUE TITLE
apollo-server-express doesn't export a server.on function

### ISSUE THREAD

**Issue by jakec-dev** [jakec-dev/travel-planner-server#45]
Hey @HcroakerDev, I've updated the server to use apollo-server-express and so far everything is working correctly, however I can't close the db connection using 

```js
server.on("close", () => {
  db.disconnect();
});
```
as it throws an error `TypeError: server.on is not a function`. This seems to be because `server` is now being created using the apollo-server-express wrapper, which doesn't export the express `on` method.

I've tried adapting the code to below which seems to me like it should work, however it throws the same error.

```js
const main = async () => {
  // ...
  return app.listen(app.get("port"), () => {
    console.log(
      `Server ready at http://localhost:${port}${server.graphqlPath}`
    );
  });
};

const expressServer = main();

expressServer.on("close", () => {
  db.disconnect();
});
```

Relevant code here: https://github.com/jakec-dev/travel-planner-server/blob/c7bb82c60e0eacf54de6c0e54c2ee678262fcb0f/src/server.js#L1-L38

I've spent a couple hours already trying to find a workaround for this but have come up empty so far, so I thought I'd follow your recommendation of asking for help if I spend more than 2 hours on a single issue. Do you have any suggestions for how I can workaround this?

**Comment by HcroakerDev** [comment]
@jakec-dev I believe you should be able to use this https://www.apollographql.com/docs/apollo-server/api/plugin/drain-http-server

Hopefully with this, you should then be able to do this:

```Javascript
httpServer.on('close', () => {
  db.disconnect();
})
```

Let me know how you go 😄

**Comment by jakec-dev** [comment]
Thanks mate. I tried the below code however it's still giving the same error. I also tried it with `server.on` and `app.on`, all of them give the same `TypeError: Cannot read properties of undefined (reading 'on')` error.

```js
const express = require("express");
const cors = require("cors");
const { ApolloServer } = require("apollo-server-express");
const { ApolloServerPluginDrainHttpServer } = require("apollo-server-core");
const http = require("http");
const typeDefs = require("./graphql/schema");
const resolvers = require("./graphql/resolvers");
const db = require("./utils/db");
const itemsRoutes = require("./routes/itemsRoutes");
const apiDocsRoutes = require("./routes/apiDocsRoutes");

const PORT = process.env.PORT || 5000;

const startApolloServer = async () => {
  const app = express();
  app.set("port", PORT);
  app.use(cors());
  app.use(express.json());
  app.use("/items", itemsRoutes);
  app.use("/api-docs", apiDocsRoutes);

  const httpServer = http.createServer(app);

  const server = new ApolloServer({
    typeDefs,
    resolvers,
    plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
  });
  await server.start();

  server.applyMiddleware({ app });
  await new Promise((resolve) => httpServer.listen({ port: PORT }, resolve));
  console.log(
    `🚀 Server ready at http://localhost:${PORT}${server.graphqlPath}`
  );

  return { httpServer, server, app };
};

const { httpServer, server } = startApolloServer();

httpServer.on("close", () => {
  db.disconnect();
});

module.exports = server;

```

**Comment by HcroakerDev** [comment]
There's an issue with your code. I'll give you a little hint...it has to do with Promises.

**Comment by jakec-dev** [comment]
Fixed by #46
